command/pr_review.go

```diff
@@ -125,10 +133,123 @@ func prReview(cmd *cobra.Command, args []string) error {
 		}
 	}
 
+	if input == nil {
+		input, err = reviewSurvey(cmd)
+		if err != nil {
+			return err
+		}
+		if input == nil && err == nil {
+			// Cancelled.
+			return nil
+		}
+	}
+
 	err = api.AddReview(apiClient, pr, input)
 	if err != nil {
 		return fmt.Errorf("failed to create review: %w", err)
 	}
 
 	return nil
 }
+
+func reviewSurvey(cmd *cobra.Command) (*api.PullRequestReviewInput, error) {
+	editorCommand, err := determineEditor(cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	typeAnswers := struct {
+		ReviewType string
+	}{}
+	typeQs := []*survey.Question{
+		{
+			Name: "reviewType",
+			Prompt: &survey.Select{
+				Message: "What kind of review do you want to give?",
+				Options: []string{
+					"Comment",
+					"Approve",
+					"Request changes",
+					"Cancel",
+				},
+			},
+		},
+	}
+
+	err = SurveyAsk(typeQs, &typeAnswers)
+	if err != nil {
+		return nil, err
+	}
+
+	reviewState := api.ReviewComment
+
+	switch typeAnswers.ReviewType {
+	case "Approve":
+		reviewState = api.ReviewApprove
+	case "Request Changes":
+		reviewState = api.ReviewRequestChanges
+	case "Cancel":
+		return nil, nil
+	}
+
+	bodyAnswers := struct {
+		Body string
+	}{}
+
+	bodyQs := []*survey.Question{
+		&survey.Question{
+			Name: "body",
+			Prompt: &surveyext.GhEditor{
+				EditorCommand: editorCommand,
+				Editor: &survey.Editor{
+					Message:  "Review body",
+					FileName: "*.md",
+				},
+			},
+		},
+	}
+
+	err = SurveyAsk(bodyQs, &bodyAnswers)
+	if err != nil {
+		return nil, err
+	}
+
+	if bodyAnswers.Body == "" && (reviewState == api.ReviewComment || reviewState == api.ReviewRequestChanges) {
+		return nil, errors.New("this type of review cannot be blank")
+	}
+
+	if len(bodyAnswers.Body) > 0 {
+		out := colorableOut(cmd)
+		renderedBody, err := utils.RenderMarkdown(bodyAnswers.Body)
+		if err != nil {
+			return nil, err
+		}
+
+		fmt.Fprintf(out, "Got:\n%s", renderedBody)
```

Comment:
Was printing the preview of the review body intended in the final product? We don't do that in `issue/pr create`, so doing it here looks odd. Also, the user has just seen this text because they've authored it in their text editor, so what would be the intended purpose of printing it out right after?

Reply:
we've discussed doing it for pr/issue as a review step. I thought this was a good opportunity to try it out.